src/app/directives/restrict-special-char.directive.ts

```diff
@@ -0,0 +1,25 @@
+import { Directive, HostListener, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appRestrictSpecialChars]'
+})
+export class RestrictSpecialCharDirective {
+  @Input() restrictedCharCodes: any[];
+
+  constructor() { }
+
+  @HostListener('keypress') onkeypress(e) {
+    const event = e || window.event;
+    if (event) {
+      return this.isRestrictedChar(event);
+    }
+  }
+
+  isRestrictedChar(event) {
+    if (this.restrictedCharCodes?.length === 0) {
```

Comment:
Is this check really required?

Reply:
Yes, it is needed as `restrictedCharCodes` is an array and it's value can be passed as `[]`